lib/usagereporter/teleport/types.go

```diff
@@ -1782,8 +1782,10 @@ func ConvertUsageEvent(event *usageeventsv1.UsageEventOneOf, userMD UserMetadata
 		}
 		return ret, nil
 	case *usageeventsv1.UsageEventOneOf_AccessListGrantsToUser:
+		// This event is emitted both as an one-off event as well as an aggregated
+		// user activity record report.
 		ret := &AccessListGrantsToUserEvent{
-			UserName:                    userMD.Username,
+			UserName:                    e.AccessListGrantsToUser.GetUserName(),
```

Comment:
I am assuming we always wanted to record the username for whom the role/trait was granted rather than the user on whose context the event was emitted. During the login event, the value of `userMD.Username` is the teleport cluster name.

Reply:
I believe usernames should always be anonymized, as they may contain sensitive PII, such as email addresses.


> I am assuming we always wanted to record the username for whom the role/trait was granted rather than the user on whose context the event was emitted

👍

Reply:
@smallinsky they are always anonymized, whatever value we provide here. Anonymization is handled by the `Anonymized` method which is called right before the final submission. See https://github.com/gravitational/teleport/blob/master/lib/usagereporter/teleport/types.go#L847

Previously, we were emitting anonymized Teleport cluster ID as username value for access list grant event. So my confusion was more of if that decision was deliberate or we mistakenly did that. In this PR, we now send anonymized user name.